### ISSUE TITLE
Fallback URLs for forward compatibility

### ISSUE THREAD

**Issue by jyasskin** [WICG/webpackage#242]
Chromium would like to be able to ship these formats and then continue evolving them as they make their way through the standards process. QUIC attempts something like this by defining ["invariants"](https://tools.ietf.org/html/draft-ietf-quic-invariants-01), "properties of the QUIC transport protocol that are expected to remain unchanged over time as new versions of the protocol are developed." To try the same thing in these specifications, it might make sense to put a fallback URL in a fixed place in both the signed exchange and bundles formats.

For signed exchanges, this would be the request URL that's currently encoded as the `:url` field in the [request headers](https://tools.ietf.org/html/draft-yasskin-http-origin-signed-responses-04#section-3.2). For bundles, it would be the [`start_url`](https://www.w3.org/TR/appmanifest/#start_url-member) that's expected to be in the bundle's [manifest](https://tools.ietf.org/html/draft-yasskin-wpack-bundled-exchanges-00#section-3.2.2). In both cases, the value would be moved instead of copied to the new location, to avoid the possibility of inconsistencies.

When the client receives an unknown version of the format (either too old or too new), it would redirect to the fallback URL. Offline users would be out of luck, but online ones would still be able to get the content.

This could also be accomplished using content negotiation on the `Accept` header, but that's a [joint that might rust shut](https://www.imperialviolet.org/2016/05/16/agility.html), so we should try to make things work even if it's rusty, if possible.

Thoughts?

**Comment by nyaxt** [comment]
Sounds good to me.
By "redirect to the fallback URL", do we expect the fallback URL to be a plain content (not signed exchange or bundle)?

**Comment by twifkak** [comment]
Sounds good to me. Are there any strategies to keep the joint oiled for the time being (e.g. through origin trial)? Are there any cross-cutting concerns (e.g. referrer, SW) that need hammering out?

**Comment by RByers** [comment]
Yes, this sounds brilliant, thank you! I hope that we can plan for doing breaking format changes over a release or two of Chrome. Ensuring that version mismatch results only in loss of performance (no ability to pre-fetch due to privacy constraints) and offline functionality instead of hard breakage is critical to enabling that.

Should we perhaps also add the higher level principle here to the "Requirements" section? I.e. signed exchanges need to be designed to plan for breaking format changes without the need for user agents to know how to decode more than one version.

**Comment by KenjiBaheux** [comment]
Cool idea.

https://github.com/WICG/webpackage/issues/242#issuecomment-402335315 : given the context, I think it makes sense to require a plain response from the fallback URL, by asking with a regular Accept header. IUC, said fallback URL would be the URL of the signed exchange's content / ..., so doing otherwise would result in a cycle.

**Comment by kinu** [comment]
+1, this sounds really neat to me.  Also +1 to mention something like this in the higher level principle section.  Reg: how this could work in play in clients/UAs I assume this'd still look like a (fancy) redirect, maybe less fancier than when the version was supported in the client.

Reg: version changes should the spec also note some recommendation about how long an old version of format should be supported by implementors?
